Framework/PythonInterface/plugins/algorithms/LRDirectBeamSort.py

```diff
@@ -43,16 +46,22 @@ def __init__(self, nxsdataToCompareWith, nxsdataToPosition):
 
         self.resultComparison = -1 if pcharge1 < pcharge2 else 1
 
-    def compareParameter(self, param, order):
+    def compareParameter(self, param, order, tolerance=None):
```

Comment:
The default value should be `0.` if you want it to be skipped by default. `bool(0.) = False` so the rest of the code would be unchanged. Otherwise, you could change change L62 to be (with the default value of `0.`)
```
if abs(_paramNexusToPosition - _paramNexusToCompareWith) <= tolerance:
```
and the code would be equivalent. The main point is that the default value should have the same type as an appropriate value.